QRCoder/QRCodeGenerator.Polynom.cs

```diff
@@ -8,21 +9,131 @@ public partial class QRCodeGenerator
         /// <summary>
         /// Represents a polynomial, which is a sum of polynomial terms.
         /// </summary>
-        private struct Polynom
+        private struct Polynom : IDisposable
         {
+            private PolynomItem[] _polyItems;
+            private int _length;
+
             /// <summary>
             /// Initializes a new instance of the <see cref="Polynom"/> struct with a specified number of initial capacity for polynomial terms.
             /// </summary>
             /// <param name="count">The initial capacity of the polynomial items list.</param>
             public Polynom(int count)
             {
-                this.PolyItems = new List<PolynomItem>(count);
+                _length = 0;
+                _polyItems = Allocate(count);
+            }
+
+            /// <summary>
+            /// Adds a polynomial term to the polynomial.
+            /// </summary>
+            public void Add(PolynomItem item)
+            {
+                EnsureCapacity(_length + 1);
+                _polyItems[_length++] = item;
+            }
+
+            /// <summary>
+            /// Removes the polynomial term at the specified index.
+            /// </summary>
+            public void RemoveAt(int index)
+            {
+                if (index < 0 || index >= _length)
```

Comment:
You can save one comparison (branch) with
```suggestion
                if ((uint)index >= (uint)_length)
```
If `index` is negative, then the `uint` cast will result in values `> int.MaxValue`, so effectively doing the same, but w/ less work.